test/unit/argument_parser/argument_parser_design_error_test.cpp

```diff
@@ -109,7 +109,7 @@ TEST(parse_test, parser_design_error)
     argument_parser parser10("test_parser", 1, argv);
     EXPECT_THROW(parser10.add_option(option_value, '\t', "no\n", "tab and newline don't work!"),
                  parser_design_error);
-    EXPECT_THROW(parser10.add_flag(flag_value, '\t', "no\n", "tab and newline don't work!"),
+    EXPECT_THROW(parser10.add_flag(flag_value, 'i', "no\n", "tab and newline don't work!"),
```

Comment:
Could you refactor this change to be part of the commit before? :)
Otherwise the change seems weird